test/langtools/tools/javac/annotations/typeAnnotations/referenceinfos/Driver.java

```diff
@@ -144,32 +137,98 @@ private Map<String, TypeAnnotation.Position> expectedOf(Method m) {
+        TypeAnnotation.TargetInfo p = null;
+        Label label = null;
+        switch (d.type()) {
+            case CAST -> {
+                p = TypeAnnotation.TargetInfo.ofCastExpr(null, d.typeIndex());
+            }
+            case CLASS_EXTENDS -> {
+                p = TypeAnnotation.TargetInfo.ofClassExtends(d.typeIndex());
+            }
+            case CONSTRUCTOR_INVOCATION_TYPE_ARGUMENT -> {
+                p = TypeAnnotation.TargetInfo.ofConstructorInvocationTypeArgument(null, d.typeIndex());
+            }
+            case CLASS_TYPE_PARAMETER -> {
+                p = TypeAnnotation.TargetInfo.ofClassTypeParameter(d.paramIndex());
+            }
+            case CLASS_TYPE_PARAMETER_BOUND -> {
+                p = TypeAnnotation.TargetInfo.ofClassTypeParameterBound(d.paramIndex(), d.boundIndex());
+            }
+            case EXCEPTION_PARAMETER -> {
+                p = TypeAnnotation.TargetInfo.ofExceptionParameter(d.exceptionIndex());
+            }
+            case FIELD -> {
+                p = TypeAnnotation.TargetInfo.ofField();
+            }
+            case INSTANCEOF -> {
+                p = TypeAnnotation.TargetInfo.ofInstanceofExpr(null);
+            }
+            case LOCAL_VARIABLE -> {
+                List<TypeAnnotation.LocalVarTargetInfo> table = new ArrayList<>();
+                for (int idx = 0; idx < d.lvarOffset().length; ++idx)
+                    table.add(TypeAnnotation.LocalVarTargetInfo.of(null, null, d.lvarIndex()[idx]));
+                p = TypeAnnotation.TargetInfo.ofLocalVariable(table);
+            }
+            case METHOD_FORMAL_PARAMETER -> {
+                p = TypeAnnotation.TargetInfo.ofMethodFormalParameter(d.paramIndex());
+            }
+            case METHOD_INVOCATION_TYPE_ARGUMENT -> {
+                p = TypeAnnotation.TargetInfo.ofMethodInvocationTypeArgument(null, d.typeIndex());
+            }
+            case METHOD_RECEIVER -> {
+                p = TypeAnnotation.TargetInfo.ofMethodReceiver();
+            }
+            case METHOD_RETURN -> {
+                p = TypeAnnotation.TargetInfo.ofMethodReturn();
+            }
+            case METHOD_TYPE_PARAMETER -> {
+                p = TypeAnnotation.TargetInfo.ofMethodTypeParameter(d.paramIndex());
+            }
+            case METHOD_TYPE_PARAMETER_BOUND -> {
+                p = TypeAnnotation.TargetInfo.ofMethodTypeParameterBound(d.paramIndex(), d.boundIndex());
+            }
+            case NEW -> {
+                p = TypeAnnotation.TargetInfo.ofNewExpr(null);
+            }
+            case RESOURCE_VARIABLE -> {
+                List<TypeAnnotation.LocalVarTargetInfo> table = new ArrayList<>();
+                for (int idx = 0; idx < d.lvarOffset().length; ++idx)
+                    table.add(TypeAnnotation.LocalVarTargetInfo.of(null, null, d.lvarIndex()[idx]));
+                p = TypeAnnotation.TargetInfo.ofResourceVariable(table);
+            }
```

Comment:
Filling missing labels with null is not ideal.
All such unsupported cases should directly throw an exception to indicate invalid state of the test.
BTW silent acceptance of null labels by TypeAnnotation.TargetInfo factory methods is a bug.

Reply:
FYI: I've filled [bug #8315541](https://bugs.openjdk.org/browse/JDK-8315541)